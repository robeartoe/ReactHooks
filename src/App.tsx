/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from 'react';
import './App.css';
import useDimensions from './util/useDimensions';

function App() {
  const [renders, setRenders] = useState<number>(0);
  const {height, width}  = useDimensions();

  useEffect(() => {
    setRenders(renders + 1);
  }, [height, width])

  return (
    <div className="App">
      <header className="App-header">
          Windows Dimensions has changed!
          <br/>
          <code>{height} x {width}</code>
          <br/>
          <code>{renders} Re-Renders</code>
      </header>
    </div>
  );
}

export default App;
