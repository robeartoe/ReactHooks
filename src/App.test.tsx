import React from 'react';
import App from './dimensionsHook';
import renderer, { act } from 'react-test-renderer';
import { mount } from 'enzyme';



describe('App', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(<App />);
  });
  it('Should render correctly - Snapshot', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have render of 1 at start', () => {
    const renderSpan = wrapper.find('#renders');
    expect(Number(renderSpan.text())).toEqual(1);
  });


  it('Should have width & height of: ', () => {
    const {innerWidth: width, innerHeight: height} = window;
    expect(Number(wrapper.find('#width').text())).toEqual(width);
    expect(Number(wrapper.find('#height').text())).toEqual(height);

  });

});


// const resizeWindow = (x, y) => {
//   window.innerWidth = x;
//   window.innerHeight = y;
//   window.dispatchEvent(new Event('resize'));
// }

describe('App - Resize', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App />);
  });
  it('Should have rendered once more.', async () => {
    // First Render
    const renderSpan = wrapper.find('#renders');
    expect(Number(renderSpan.text())).toEqual(1);
    // Resize:
    // Change the viewport to 500px.
    global.innerWidth = 500;
    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize'));
    expect(Number(renderSpan.text())).toEqual(2);
  });
  it('Should have different width.', () => {
    let {innerWidth: width} = window;
    expect(Number(wrapper.find('#width').text())).toEqual(width);
    // Resize:
    // Change the viewport to 500px.
    global.innerWidth = 500;
    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize'));
    expect(Number(wrapper.find('#width').text())).toEqual(500);

  });

  it('Should have different height.', () => {
    let { innerHeight: height } = window;
    expect(Number(wrapper.find('#height').text())).toEqual(height);
    // Resize:
    // Change the viewport to 500px.
    global.innerHeight = 500;
    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize'));
    expect(Number(wrapper.find('#height').text())).toEqual(500);
  });
});