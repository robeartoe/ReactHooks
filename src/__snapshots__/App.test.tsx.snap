// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render correctly - Snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    Windows Dimensions has changed!
    <br />
    <code>
      <span
        id="height"
      >
        768
      </span>
      x
      <span
        id="width"
      >
        1024
      </span>
    </code>
    <br />
    <code>
      <span
        id="renders"
      >
        0
      </span>
       Re-Renders
    </code>
  </header>
</div>
`;
